src/axolotl/utils/config/__init__.py

```diff
@@ -94,6 +100,33 @@ def normalize_config(cfg):
         if cfg.bf16:
             cfg.fp16 = True
         cfg.bf16 = False
+    elif cfg.device.startswith("npu"):
+        if cfg.load_in_8bit or cfg.load_in_4bit:
+            LOG.warning(
+                "Quantification is currently not supported in Ascend npu, disabling for this configuration."
+            )
+            cfg.load_in_8bit = False
+            cfg.load_in_4bit = False
+
+        if cfg.tf32:
+            LOG.warning(
+                "tf32 dtype is currently not supported in Ascend npu, disabling for this configuration."
+            )
+            cfg.tf32 = False
+
+        if cfg.flash_attention:
```

Comment:
How about `xformers_attention` , `sdp_attention`, and `s2_attention`?

Reply:
These are not supported in Ascend NPU either